src/platform/plugins/private/inspect_component/server/lib/codeowners/get_component_codeowners.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { existsSync, readFileSync } from 'fs';
+import { join, sep } from 'path';
+import { REPO_ROOT } from '@kbn/repo-info';
+
+/**
+ * Get the list of codeowners for a given file path by reading the CODEOWNERS file in the repository.
+ * It matches the most specific path.
+ * @param {string} path The file path to look up codeowners for, relative to the repository root.
+ * @returns {string[]} An array of codeowners responsible for the given path.
+ */
+export function getComponentCodeowners(path: string): string[] {
```

Comment:
might be worth to just read this file to memory with a reference we hold on to so we aren't doing this every time we get a new request, then attempt memoizing this function so that already encountered paths don't require a lookup